### PR TITLE
feat: Add kingfisher Package

### PR DESCRIPTION
**Pull request by RaniaZeid**

### CODE REVIEW

**Comment by RaniaZeid** [comment]
@ahmednaagy @ahmdyasser  @MarwaEbrahem  @HebaMad  @claramounir  review please

File: Dokan/Dokan/Extension/UIImage+Extension.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+//
+//  Created by raniazeid on 25/07/2022.
+//
+
+import Foundation
+import Kingfisher
+import UIKit
+
+
+extension UIImageView{
```
Comment (oradwanomar) [inline Dokan/Dokan/Extension/UIImage+Extension.swift]:
```suggestion
protocol ImageSetable {
      func setImage(with urlString: String , placeholderString : String)
}

extension UIImageView: ImageSetable {
```

**Review by oradwanomar — APPROVED** [review]

File: Dokan/Dokan/Extension/UIImage+Extension.swift (hunk 1 above)
Comment (ahmdmhasn) [inline Dokan/Dokan/Extension/UIImage+Extension.swift]:
```suggestion
extension UIImageView {
```

File: Dokan/Dokan/Extension/UIImage+Extension.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    func setImage(with urlString: String , placeholderString : String ) {
+         let url = URL(string: urlString)
+         self.kf.setImage(with: url,
+                          placeholder: #imageLiteral(resourceName: placeholderString),
+                          options: [.transition(.fade(0.5))],
+                          progressBlock: nil, completionHandler: { [weak self](result) in
+                             switch result {
+                             case .failure:
+                                 self?.image =  #imageLiteral(resourceName: "personImage")
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Extension/UIImage+Extension.swift]: What is person image?!

Comment (RaniaZeid) [reply Dokan/Dokan/Extension/UIImage+Extension.swift]: added by mistakes I will edit it

**Comment by RaniaZeid** [comment]
@ahmdmhasn  I fixed the isuue u can review last commit

File: Dokan/Dokan/Extension/UIImage+Extension.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import Foundation
+import Kingfisher
+import UIKit
+
+protocol ImageSetable {
+      func setImage(with urlString: String , placeholderString : String)
+}
+extension UIImageView : ImageSetable {
+
+    func setImage(with urlString: String , placeholderString : String ) {
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Extension/UIImage+Extension.swift]: Placeholder should be `UIImage` type

File: Dokan/Dokan/Extension/UIImage+Extension.swift [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+         self.kf.setImage(with: url,
+                          placeholder: #imageLiteral(resourceName: placeholderString),
+                          options: [.transition(.fade(0.5))],
+                          progressBlock: nil, completionHandler: { [weak self](result) in
+                             switch result {
+                             case .failure:
+                                 self?.image =  #imageLiteral(resourceName: placeholderString)
+                             default:
+                                 break
+                             }
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Extension/UIImage+Extension.swift]: Block is not needed since the library automatically does this for you.

**Review by ahmdmhasn — COMMENTED** [review]
Please rebuild the project to fix the indentation.

**Review by ahmednaagy — APPROVED** [review]

File: Dokan/Dokan.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -252,8 +279,12 @@
 		AA21B43428537E0400E4F217 /* Dokan */ = {
 			isa = PBXGroup;
 			children = (
+
+
 				432EF5D5288F2B340073FB1D /* Localization */,
+
 				99CC7307288E2FE400503D86 /* Colors.xcassets */,
+
```
Comment (ahmdmhasn) [inline Dokan/Dokan.xcodeproj/project.pbxproj]: Spaces here aren't supposed to exist.

File: Dokan/Dokan.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -210,6 +236,7 @@
 			path = InfoSellerView;
 			sourceTree = "<group>";
 		};
+
```
Comment (ahmdmhasn) [inline Dokan/Dokan.xcodeproj/project.pbxproj]: Spaces here aren't supposed to exist.

File: Dokan/Dokan.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -147,6 +161,18 @@
...
+		5120E6AD288F2F3800BCC7DE /* Extension */ = {
+			isa = PBXGroup;
+			children = (
+				5120E6AE288F2F6400BCC7DE /* UIImage+Extension.swift */,
+			);
+			path = Extension;
+			sourceTree = "<group>";
+		};
+
+
```
Comment (ahmdmhasn) [inline Dokan/Dokan.xcodeproj/project.pbxproj]: White spaces here aren't supposed to exist.

File: Dokan/Dokan.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -120,6 +129,11 @@
 				856E36B928893C9A00735EF2 /* UIDokan in Frameworks */,
 				AAE1A1BF285E1A3A00A4EBB0 /* Networking in Frameworks */,
 				AAA2E180285CBFCA00EAD189 /* FirebaseAnalytics in Frameworks */,
+
+				5120E6AC288F2E8600BCC7DE /* Kingfisher in Frameworks */,
+				AAE1A1C1285E1A3D00A4EBB0 /* UIDokan in Frameworks */,
+
+
```
Comment (ahmdmhasn) [inline Dokan/Dokan.xcodeproj/project.pbxproj]: White spaces here aren't supposed to exist.

File: Dokan/Dokan.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -80,6 +88,7 @@
 		43D4BF0A2891A5C50023190F /* FeaturedProductCollectionViewCellTests.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = FeaturedProductCollectionViewCellTests.swift; sourceTree = "<group>"; };
 		856E36AE2889229B00735EF2 /* InfoSellerView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = InfoSellerView.swift; sourceTree = "<group>"; };
 		856E36B0288922EF00735EF2 /* InfoSellerView.xib */ = {isa = PBXFileReference; lastKnownFileType = file.xib; path = InfoSellerView.xib; sourceTree = "<group>"; };
+
```
Comment (ahmdmhasn) [inline Dokan/Dokan.xcodeproj/project.pbxproj]: White spaces here aren't supposed to exist.

File: Dokan/Dokan.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -18,11 +18,14 @@
...
+		5120E6AC288F2E8600BCC7DE /* Kingfisher in Frameworks */ = {isa = PBXBuildFile; productRef = 5120E6AB288F2E8600BCC7DE /* Kingfisher */; };
+		5120E6AF288F2F6400BCC7DE /* UIImage+Extension.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5120E6AE288F2F6400BCC7DE /* UIImage+Extension.swift */; };
 		43D4BF0928918F9A0023190F /* FeaturedProductTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 43D4BF0828918F9A0023190F /* FeaturedProductTests.swift */; };
 		43D4BF0B2891A5C50023190F /* FeaturedProductCollectionViewCellTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 43D4BF0A2891A5C50023190F /* FeaturedProductCollectionViewCellTests.swift */; };
 		856E36AF2889229B00735EF2 /* InfoSellerView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 856E36AE2889229B00735EF2 /* InfoSellerView.swift */; };
 		856E36B1288922EF00735EF2 /* InfoSellerView.xib in Resources */ = {isa = PBXBuildFile; fileRef = 856E36B0288922EF00735EF2 /* InfoSellerView.xib */; };
 		856E36B928893C9A00735EF2 /* UIDokan in Frameworks */ = {isa = PBXBuildFile; productRef = AAE1A1C0285E1A3D00A4EBB0 /* UIDokan */; };
+
```
Comment (ahmdmhasn) [inline Dokan/Dokan.xcodeproj/project.pbxproj]: White spaces here aren't supposed to exist.

File: Dokan/Dokan.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -70,6 +73,11 @@
 		276889DB287417730091ACEB /* RegisterViewController.xib */ = {isa = PBXFileReference; lastKnownFileType = file.xib; path = RegisterViewController.xib; sourceTree = "<group>"; };
 		276889DE28741F750091ACEB /* RegisterViewModel.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = RegisterViewModel.swift; sourceTree = "<group>"; };
 		276889E0287420980091ACEB /* RegisterViewModelType.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = RegisterViewModelType.swift; sourceTree = "<group>"; };
+
+		276889E228742A4C0091ACEB /* RegisterViewControllerTests.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; name = RegisterViewControllerTests.swift; path = DokanTests/Scenes/Registratio9nTest/RegisterViewControllerTests.swift; sourceTree = SOURCE_ROOT; };
+
+		5120E6AE288F2F6400BCC7DE /* UIImage+Extension.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = "UIImage+Extension.swift"; sourceTree = "<group>"; };
+
```
Comment (ahmdmhasn) [inline Dokan/Dokan.xcodeproj/project.pbxproj]: White spaces here aren't supposed to exist.

File: Dokan/Dokan/Extension/UIImage+Extension.swift (hunk 3 above)
Comment (ahmdmhasn) [reply Dokan/Dokan/Extension/UIImage+Extension.swift]: @RaniaZeid
Any update?

File: Dokan/Dokan/Extension/UIImage+Extension.swift (hunk 4 above)
Comment (ahmdmhasn) [reply Dokan/Dokan/Extension/UIImage+Extension.swift]: @RaniaZeid ?

**Comment by RaniaZeid** [comment]
@ahmdmhasn  u can check the update

**Comment by RaniaZeid** [comment]
another one open #99

**Pull request closed without merging**